cli/azd/pkg/project/ai_helper.go

```diff
@@ -0,0 +1,713 @@
+}
+
+// NewAiHelper creates a new instance of AiHelper
+func NewAiHelper(
+	env *environment.Environment,
+	clock clock.Clock,
+	credentialProvider account.SubscriptionCredentialProvider,
+	pythonBridge ai.PythonBridge,
+	workspacesClient *armmachinelearning.WorkspacesClient,
+	envContainersClient *armmachinelearning.EnvironmentContainersClient,
+	envVersionsClient *armmachinelearning.EnvironmentVersionsClient,
+	modelContainersClient *armmachinelearning.ModelContainersClient,
+	modelVersionsClient *armmachinelearning.ModelVersionsClient,
+	endpointsClient *armmachinelearning.OnlineEndpointsClient,
+	deploymentsClient *armmachinelearning.OnlineDeploymentsClient,
+) AiHelper {
+	return &aiHelper{
+		env:                   env,
+		clock:                 clock,
+		pythonBridge:          pythonBridge,
+		workspacesClient:      workspacesClient,
+		envContainersClient:   envContainersClient,
+		envVersionsClient:     envVersionsClient,
+		modelContainersClient: modelContainersClient,
+		modelVersionsClient:   modelVersionsClient,
+		endpointsClient:       endpointsClient,
+		deploymentsClient:     deploymentsClient,
+	}
+}
+
+// RequiredExternalTools returns the required external tools for the AiHelper
+func (a *aiHelper) RequiredExternalTools(ctx context.Context) []tools.ExternalTool {
+	return a.pythonBridge.RequiredExternalTools(ctx)
+}
+
+// Initialize initializes the AiHelper
+func (a *aiHelper) Initialize(ctx context.Context) error {
+	if a.initialized {
+		return nil
+	}
+
+	if err := a.pythonBridge.Initialize(ctx); err != nil {
+		return err
+	}
+
+	a.initialized = true
+	return nil
+}
+
+// EnsureWorkspace ensures that the workspace exists
+func (a *aiHelper) EnsureWorkspace(
+	ctx context.Context,
+	scope *ai.Scope,
+) error {
+	workspaceName := scope.Workspace()
+
+	workspaceResponse, err := a.workspacesClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		workspaceName,
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	if *workspaceResponse.Workspace.Name != workspaceName {
+		return err
+	}
+
+	return nil
+}
+
+// CreateEnvironmentVersion creates a new environment version
+func (a *aiHelper) CreateEnvironmentVersion(
+	ctx context.Context,
+	scope *ai.Scope,
+	serviceConfig *ServiceConfig,
+	config *ai.ComponentConfig,
+) (*armmachinelearning.EnvironmentVersion, error) {
+	yamlFilePath := filepath.Join(serviceConfig.Path(), config.Path)
+	_, err := os.Stat(yamlFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	environmentName, err := config.Name.Envsubst(a.env.Getenv)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing environment name value: %w", err)
+	}
+
+	if environmentName == "" {
+		environmentName = fmt.Sprintf("%s-environment", serviceConfig.Name)
+	}
+
+	nextVersion := "1"
+	envContainerResponse, err := a.envContainersClient.Get(
+		ctx,
+		scope.ResourceGroup(),
+		scope.Workspace(),
+		environmentName,
+		nil,
+	)
+	if err == nil {
+		nextVersion = *envContainerResponse.Properties.NextVersion
+	}
```

Comment:
Do we want to ignore /all/ errors here?  My guess is that we are okay with an error if the resource is not found, but if there was some other network or auth error we should fail here?

Reply:
We can explicitly check for a 404 error.  In this case that means this model was never previously published and we'll start with `1`.